homeassistant/components/solarlog/const.py

```diff
@@ -17,83 +30,190 @@
-    ],
-    "total_power": ["totalPOWER", "total power", "Wp", "mdi:solar-power"],
-    "alternator_loss": [
-        "alternatorLOSS",
-        "alternator loss",
-        POWER_WATT,
-        "mdi:solar-power",
-    ],
-    "capacity": ["CAPACITY", "capacity", PERCENTAGE, "mdi:solar-power"],
-    "efficiency": [
-        "EFFICIENCY",
-        "efficiency",
-        f"% {POWER_WATT}/{POWER_WATT}p",
-        "mdi:solar-power",
-    ],
-    "power_available": [
-        "powerAVAILABLE",
-        "power available",
-        POWER_WATT,
-        "mdi:solar-power",
-    ],
-    "usage": ["USAGE", "usage", None, "mdi:solar-power"],
-}
+
+@dataclass
+class SolarlogRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    json_key: str
+
+
+@dataclass
+class SolarLogSensorEntityDescription(
+    SensorEntityDescription, SolarlogRequiredKeysMixin
+):
+    """Describes Solarlog sensor entity."""
+
+
+SENSOR_TYPES: tuple[SolarLogSensorEntityDescription, ...] = (
+    SolarLogSensorEntityDescription(
+        key="time",
+        json_key="TIME",
+        name="last update",
+        device_class=DEVICE_CLASS_TIMESTAMP,
+    ),
+    SolarLogSensorEntityDescription(
+        key="power_ac",
+        json_key="powerAC",
+        name="power AC",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=POWER_WATT,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SolarLogSensorEntityDescription(
+        key="power_dc",
+        json_key="powerDC",
+        name="power DC",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=POWER_WATT,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SolarLogSensorEntityDescription(
+        key="voltage_ac",
+        json_key="voltageAC",
+        name="voltage AC",
+        native_unit_of_measurement=ELECTRIC_POTENTIAL_VOLT,
+        device_class=DEVICE_CLASS_VOLTAGE,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SolarLogSensorEntityDescription(
+        key="voltage_dc",
+        json_key="voltageDC",
+        name="voltage DC",
+        native_unit_of_measurement=ELECTRIC_POTENTIAL_VOLT,
+        device_class=DEVICE_CLASS_VOLTAGE,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SolarLogSensorEntityDescription(
+        key="yield_day",
+        json_key="yieldDAY",
+        name="yield day",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SolarLogSensorEntityDescription(
+        key="yield_yesterday",
+        json_key="yieldYESTERDAY",
+        name="yield yesterday",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+    ),
+    SolarLogSensorEntityDescription(
+        key="yield_month",
+        json_key="yieldMONTH",
+        name="yield month",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        state_class=STATE_CLASS_MEASUREMENT,
```

Comment:
kWh is not a current value/measurement, it is an aggregation of today's usage in this case. Thus it cannot be a measurement. `STATE_CLASS_TOTAL_INCREASING` is more fitting in this case, but this is covered by `yield_total`.

So I would suggest removing it here

```suggestion
```